de.metas.contracts/src/main/java/de/metas/contracts/pricing/trade_margin/CustomerTradeMarginPricingRule.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contracts.pricing.trade_margin;
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.service.IBPartnerBL;
+import de.metas.contracts.commission.Beneficiary;
+import de.metas.contracts.commission.commissioninstance.businesslogic.CommissionContract;
+import de.metas.contracts.commission.commissioninstance.businesslogic.CommissionInstance;
+import de.metas.contracts.commission.commissioninstance.businesslogic.CommissionPoints;
+import de.metas.contracts.commission.commissioninstance.services.CommissionInstanceService;
+import de.metas.contracts.commission.commissioninstance.services.CommissionPointsService;
+import de.metas.contracts.commission.commissioninstance.services.CreateForecastCommissionInstanceRequest;
+import de.metas.logging.LogManager;
+import de.metas.money.Money;
+import de.metas.money.MoneyService;
+import de.metas.pricing.IPricingContext;
+import de.metas.pricing.IPricingResult;
+import de.metas.pricing.rules.IPricingRule;
+import de.metas.product.ProductPrice;
+import de.metas.quantity.Quantitys;
+import de.metas.util.Services;
+import org.compiere.SpringContextHolder;
+import org.slf4j.Logger;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+public class CustomerTradeMarginPricingRule implements IPricingRule
+{
+	private static final Logger logger = LogManager.getLogger(CustomerTradeMarginPricingRule.class);
+
+	private final CustomerTradeMarginService customerTradeMarginService = SpringContextHolder.instance.getBean(CustomerTradeMarginService.class);
+
+	private final CommissionInstanceService commissionInstanceService = SpringContextHolder.instance.getBean(CommissionInstanceService.class);
+
+
+	@Override public boolean applies(IPricingContext pricingCtx, IPricingResult result)
+	{
+		if ( !result.isCalculated() )
+		{
+			logger.debug("Not applying due to missing calculated price!");
```

Comment:
As of very recently we register a dedicated loggable while the pricing engine runs.
So there is something to gain by
```suggestion
			Loggables.withLogger(logger, Level.DEBUG).addLog("Not applying due to missing calculated price!");
```